internal/osbuild/gcp_guest_agent_conf_stage.go

```diff
@@ -0,0 +1,118 @@
+package osbuild
+
+import (
+	"fmt"
+)
+
+type GcpGuestAgentConfigScopeValue string
+
+const (
+	GcpGuestAgentConfigScopeDistro   GcpGuestAgentConfigScopeValue = "distro"
+	GcpGuestAgentConfigScopeInstance GcpGuestAgentConfigScopeValue = "instance"
+)
+
+type GcpGuestAgentConfigAccounts struct {
+	DeprovisionRemove *bool    `json:"deprovision_remove,omitempty"`
+	Groups            []string `json:"groups,omitempty"`
+	UseraddCmd        string   `json:"useradd_cmd,omitempty"`
+	UserdelCmd        string   `json:"userdel_cmd,omitempty"`
+	UsermodCmd        string   `json:"usermod_cmd,omitempty"`
+	GpasswdAddCmd     string   `json:"gpasswd_add_cmd,omitempty"`
+	GpasswdRemoveCmd  string   `json:"gpasswd_remove_cmd,omitempty"`
+	GroupaddCmd       string   `json:"groupadd_cmd,omitempty"`
+}
+
+type GcpGuestAgentConfigDaemons struct {
+	AccountsDaemon  *bool `json:"accounts_daemon,omitempty"`
+	ClockSkewDaemon *bool `json:"clock_skew_daemon,omitempty"`
+	NetworkDaemon   *bool `json:"network_daemon,omitempty"`
+}
+
+type GcpGuestAgentConfigInstanceSetup struct {
+	HostKeyTypes     []string `json:"host_key_types,omitempty"`
+	OptimizeLocalSsd *bool    `json:"optimize_local_ssd,omitempty"`
+	NetworkEnabled   *bool    `json:"network_enabled,omitempty"`
+	SetBotoConfig    *bool    `json:"set_boto_config,omitempty"`
+	SetHostKeys      *bool    `json:"set_host_keys,omitempty"`
+	SetMultiqueue    *bool    `json:"set_multiqueue,omitempty"`
+}
+
+type GcpGuestAgentConfigIpForwarding struct {
+	EthernetProtoId   string `json:"ethernet_proto_id,omitempty"`
+	IpAliases         *bool  `json:"ip_aliases,omitempty"`
+	TargetInstanceIps *bool  `json:"target_instance_ips,omitempty"`
+}
+
+type GcpGuestAgentConfigMetadataScripts struct {
+	DefaultShell string `json:"default_shell,omitempty"`
+	RunDir       string `json:"run_dir,omitempty"`
+	Startup      *bool  `json:"startup,omitempty"`
+	Shutdown     *bool  `json:"shutdown,omitempty"`
+}
+
+type GcpGuestAgentConfigNetworkInterfaces struct {
+	Setup        *bool  `json:"setup,omitempty"`
+	IpForwarding *bool  `json:"ip_forwarding,omitempty"`
+	DhcpCommand  string `json:"dhcp_command,omitempty"`
+}
+
+type GcpGuestAgentConfig struct {
+	GcpGuestAgentConfigAccounts          *GcpGuestAgentConfigAccounts          `json:"Accounts,omitempty"`
+	GcpGuestAgentConfigDaemons           *GcpGuestAgentConfigDaemons           `json:"Daemons,omitempty"`
+	GcpGuestAgentConfigInstanceSetup     *GcpGuestAgentConfigInstanceSetup     `json:"InstanceSetup,omitempty"`
+	GcpGuestAgentConfigIpForwarding      *GcpGuestAgentConfigIpForwarding      `json:"IpForwarding,omitempty"`
+	GcpGuestAgentConfigMetadataScripts   *GcpGuestAgentConfigMetadataScripts   `json:"MetadataScripts,omitempty"`
+	GcpGuestAgentConfigNetworkInterfaces *GcpGuestAgentConfigNetworkInterfaces `json:"NetworkInterfaces,omitempty"`
```

Comment:
I'm sorry, but you probably missed these member names when you were removing the `GcpGuestAgentConfig` prefix from names in other structures.